core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -506,15 +539,15 @@ private boolean insertAsSelect(
                 }
             }
 
-            final long refreshFinishTimestamp = microsecondClock.getTicks();
+            if (commitCount == 0) {
```

Comment:
nit: this branch is never taken in the current tests. That's because the only way to reach it now is to have the iterator return no intervals. So, it could be replaced with an assertion.